### PR TITLE
Feat/last seen by env UI

### PR DESCRIPTION
**Pull request by andreas-unleash**
Implements last seen by environment UI

Closes # [1-1182](https://linear.app/unleash/issue/1-1182/ui-for-last-seen-per-environment)

Screenshot 2023-08-08 at 10 30 56
Screenshot 2023-08-08 at 10 31 24

### CODE REVIEW

File: frontend/src/component/common/Table/cells/FeatureSeenCell/LastSeenTooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+interface ILastSeenTooltipProps {
+    environments?: IEnvironments[];
+    className?: string;
+    sx?: SxProps<Theme>;
+}
+
+const useLastSeeneColor = () => {
+    const theme = useTheme();
+
+    return (unit?: string): string => {
```
Comment (Tymek) [inline frontend/src/component/common/Table/cells/FeatureSeenCell/LastSeenTooltip.tsx:54]: Duplicate with `useFeatureColor`?

File: frontend/src/component/common/Table/cells/FeatureSeenCell/LastSeenTooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    return (
+        <StyledDescription {...rest}>
+            <StyledDescriptionHeader sx={{ mb: 0 }}>
+                Last usage reported
+            </StyledDescriptionHeader>
+            <StyledDescriptionSubHeader>
+                Usage is reported from connected applications through metrics
+            </StyledDescriptionSubHeader>
+            {environments &&
+                environments?.map(({ name, lastSeenAt }) => (
```
Comment (kwasniew) [inline frontend/src/component/common/Table/cells/FeatureSeenCell/LastSeenTooltip.tsx:90]: redundant ?. It's already checked with environments &&

File: frontend/src/component/common/Table/cells/FeatureSeenCell/LastSeenTooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                                    title=""
+                                    live={false}
+                                    formatter={(
+                                        value: number,
+                                        unit: string,
+                                        suffix: string
+                                    ) => {
+                                        return (
+                                            <div
+                                                style={{
```
Comment (kwasniew) [inline frontend/src/component/common/Table/cells/FeatureSeenCell/LastSeenTooltip.tsx:109]: inline styling

File: frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    const environmentWithMetrics = environments.filter(
+        (environment: IEnvironments) => environment.lastSeenAt != null
+    );
+    console.log(feature.name);
+    return (
+        <ConditionallyRender
+            condition={Boolean(environmentWithMetrics)}
+            show={
+                <TimeAgo
+                    date={feature.lastSeenAt!}
```
Comment (kwasniew) [inline frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx:117]: this is dangerous, can we do it without non-null assertions?

File: frontend/src/component/common/Table/cells/FeatureSeenCell/LastSeenTooltip.tsx (hunk 1 above)
Comment (andreas-unleash) [reply frontend/src/component/common/Table/cells/FeatureSeenCell/LastSeenTooltip.tsx:54]: sort of - gives different colors for the text

File: frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    justifyContent: 'center',
+    fontSize: theme.typography.body2.fontSize,
+    margin: '0 auto',
+}));
+
+interface IFeatureSeenCellProps {
+    feature: IFeatureToggleListItem;
+}
+
+const Wrapper: FC<{
```
Comment (kwasniew) [inline frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx:49]: Maybe TooltipContainer? Wrapper is quite generic

File: frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+            <TooltipResolver
+                variant="custom"
+                titleComponent={tooltip}
+                arrow
+                describeChild
+            >
+                <StyledBox sx={{ '&:hover': { background: color } }}>
+                    <StyledIconWrapper
+                        style={{ background: color }}
+                        data-loading
```
Comment (kwasniew) [inline frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx:64]: what is data-loading doing?

File: frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+                    >
+                        {children}
+                    </StyledIconWrapper>
+                </StyledBox>
+            </TooltipResolver>
+        </StyledContainer>
+    );
+};
+
+const useFeatureColor = () => {
```
Comment (kwasniew) [inline frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx:74]: what about grouping similar cases?
```
const useFeatureColor = () => {
    const theme = useTheme();

    return (unit?: string): [string, string] => {
        switch (unit) {
            case 'second':
            case 'minute':
            case 'hour':
            case 'day':
                return [theme.palette.seen.recent, theme.palette.success.main];

            case 'week':
            case 'weeks':
                return [
                    theme.palette.seen.inactive,
                    theme.palette.warning.main,
                ];

            case 'month':
            case 'year':
                return [theme.palette.seen.abandoned, theme.palette.error.main];

            default:
                return [theme.palette.seen.unknown, theme.palette.grey.A400];
        }
    };
};
```

File: frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx (hunk 4 above)
Comment (andreas-unleash) [reply frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx:117]: The check in the ConditionallyRender component,  verifies that there is at least one environment with lastSeen metrics, so the value of `feature.lastSeenAt` is guaranteed

Comment (kwasniew) [reply frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx:117]: hmm, in the condition we have Boolean(environmentWithMetrics) not lastSeenAt

File: frontend/src/component/common/Table/cells/FeatureSeenCell/LastSeenTooltip.tsx (hunk 1 above)
Comment (andreas-unleash) [reply frontend/src/component/common/Table/cells/FeatureSeenCell/LastSeenTooltip.tsx]: Refactored out the hook

File: frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx (hunk 4 above)
Comment (andreas-unleash) [reply frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx]: Changed

File: frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx (hunk 6 above)
Comment (andreas-unleash) [reply frontend/src/component/common/Table/cells/FeatureSeenCell/FeatureEnvironmentSeenCell.tsx]: removed

**Review by kwasniew — APPROVED** [review]

**Pull request merged**